url4c41254592_topskychen_AuthTopk_tgz-pJ8-quickhull3d_QuickHull3DTestJ8/src/utility/VerifyIndex.java

```diff
@@ -0,0 +1,105 @@
+/**
+ * 
+ */
+package utility;
+
+import java.io.BufferedInputStream;
+import java.io.DataInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.RandomAccessFile;
+import java.util.HashMap;
+import java.util.Iterator;
+
+import utility.Compare.DataOfPoint;
+import utility.geo.DataOfLine;
+
+/**
+ * @author chenqian
+ *
+ */
+public class VerifyIndex {
+
+	public static String idx = "/tmp/qchen/database/GO1.0";
+	public static RandomAccessFile rafPoint = null;
+	public static HashMap<Long, long[]> tOfPoint = null;
+	public static RandomAccessFile rafLine = null;
+	public static HashMap<Long, long[]> tOfLine = null;
+	
+	public static void loadIndex() throws IOException{
+		DataInputStream dis = new DataInputStream(new BufferedInputStream(new FileInputStream(new File(idx + ".dp.idx"))));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *dis*. The resource is referenced at lines 32, 35, 36. The resource is closed at line 38. There are other execution paths that do not contain closure statements, for example, when FilterInputStream.available throws an exception. Ensure *dis* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).